stripe/src/main/java/com/stripe/android/paymentsheet/model/ClientSecret.kt

```diff
@@ -10,6 +10,7 @@ import kotlinx.parcelize.Parcelize
  */
 internal sealed class ClientSecret : Parcelable {
     abstract val value: String
+    abstract fun createConfirmParamsFactory(): ConfirmStripeIntentParamsFactory<*>
```

Comment:
Nit: feels like something that `ClientSecret` shouldn't need to know about. Instead, can we create a `ConfirmStripeIntentParamsFactory.create(clientSecret: ClientSecret)` that will create the correct `ConfirmStripeIntentParamsFactory`?

Reply:
Yeah, I was going to do something similar to that, but put it here to take advantage of the existing abstraction and avoid adding conditionals on the creation of the dependencies that are PaymentIntent/SetupIntent specific. Maybe just have a better name, and let this be the class that defines whether we're handling a PaymentIntent or SetupIntent, holding a reference to the client secret and providing those dependencies.
Its only property will be the client secret, the methods can be static.

Reply:
```suggestion
    abstract fun createConfirmParamsFactory(): ConfirmStripeIntentParamsFactory<out ConfirmStripeIntentParams>
```

https://stackoverflow.com/questions/44298702/what-is-out-keyword-in-kotlin

Reply:
Added `out` to the base class definition.

Reply:
Continuing my previous comment `createConfirmParamsFactory` doesn't modify anything on `ClientSecret` but it wouldn't make sense to put it in a companion object or we'd lose the ability to override it on the subclasses.

Reply:
hmm could you also change the generic type to `ConfirmStripeIntentParams` for readability purposes?
```abstract fun createConfirmParamsFactory(): ConfirmStripeIntentParamsFactory<ConfirmStripeIntentParams>```

Reply:
Sure, done.